wgpu/backends/wgpu_native/_api.py

```diff
@@ -2050,10 +2050,28 @@ def set_bind_group(
         self,
         index,
         bind_group,
-        dynamic_offsets_data,
-        dynamic_offsets_data_start,
-        dynamic_offsets_data_length,
+        dynamic_offsets_data=[],
+        dynamic_offsets_data_start=None,
+        dynamic_offsets_data_length=None,
     ):
+        if (
+            dynamic_offsets_data_start is not None
+            or dynamic_offsets_data_length is not None
+        ):
+            assert (
+                dynamic_offsets_data_start is not None
+                and dynamic_offsets_data_start >= 0
+            )
+            assert (
+                dynamic_offsets_data_length is not None
+                and dynamic_offsets_data_start + dynamic_offsets_data_length
+                <= len(dynamic_offsets_data)
```

Comment:
For backwards compatibility we can allow lengths that are too large. Since Python's slicing already behaves this way, we only have to remove the test for it.